### PR TITLE
Add support api 2.0 to layer_tests

### PR DESCRIPTION
**Pull request by ishariko**
### Tickets:
 - *74832*

### CODE REVIEW

File: tests/layer_tests/common/tf_layer_test_class.py [hunk 1]
```diff
@@ -62,13 +62,20 @@ def get_framework_results(self, inputs_dict, model_path):
...
-                    input[key + ':0'] = transpose_nchw_to_nhwc(data, self.use_new_frontend)
+                if self.api_2:
+                    input.update(inputs_dict)
+                else:
+                    for key in inputs_dict.keys():
+                        data = inputs_dict.get(key)
+                        if not self.api_2:
+                            key += ':0'
+                        input[key] = transpose_nchw_to_nhwc(data, self.use_new_frontend, self.api_2)
+                        # input[key] = transpose_nchw_to_nhwc(data, self.use_new_frontend)
```
Comment (dkrupnov) [inline tests/layer_tests/common/tf_layer_test_class.py:73]: remove

File: tests/layer_tests/onnx_tests/test_loop.py [hunk 2]
```diff
@@ -270,15 +273,15 @@ def create_loop_in_loop(self):
 
     @pytest.mark.precommit
     @pytest.mark.timeout(250)
-    def test_loop_simple_precommit(self, ie_device, precision, ir_version, temp_dir):
+    def test_loop_simple_precommit(self, ie_device, precision, ir_version, temp_dir, api_2):
         if ie_device == 'GPU':
             pytest.skip('Loop not supported on GPU')
         self._test(*self.create_loop(), ie_device, precision, ir_version, temp_dir=temp_dir,
```
Comment (dkrupnov) [inline tests/layer_tests/onnx_tests/test_loop.py:279]: add api_2=api_2

File: tests/layer_tests/onnx_tests/test_loop.py [hunk 3]
```diff
@@ -270,15 +273,15 @@ def create_loop_in_loop(self):
...
                    infer_timeout=150)
 
     @pytest.mark.precommit
     @pytest.mark.timeout(250)
-    def test_loop_in_loop_simple_precommit(self, ie_device, precision, ir_version, temp_dir):
+    def test_loop_in_loop_simple_precommit(self, ie_device, precision, ir_version, temp_dir, api_2):
         pytest.skip('The model used in the test is incorrect according to ONNX standart: 70158')
         self._test(*self.create_loop_in_loop(), ie_device, precision, ir_version, temp_dir=temp_dir,
```
Comment (dkrupnov) [inline tests/layer_tests/onnx_tests/test_loop.py:286]: add api_2=api_2

File: tests/layer_tests/onnx_tests/test_lstm.py [hunk 4]
```diff
@@ -143,20 +143,26 @@ def create_lstm(self, direction: str, cell_type: str, hidden_size=128):
     @pytest.mark.timeout(250)
     @pytest.mark.parametrize('direction', ["forward", "bidirectional", "reverse"])
     @pytest.mark.parametrize('cell_type', ["LSTM", "GRU", "RNN"])
-    def test_lstm_simple_precommit(self, direction, cell_type, ie_device, precision, ir_version, temp_dir):
-        self._test(*self.create_lstm(direction, cell_type), ie_device, precision, ir_version, temp_dir=temp_dir,
+    def test_lstm_simple_precommit(self, direction, cell_type, ie_device, precision, ir_version,
+                                   temp_dir, api_2):
+        self._test(*self.create_lstm(direction, cell_type), ie_device, precision, ir_version,
+                   temp_dir=temp_dir,
```
Comment (dkrupnov) [inline tests/layer_tests/onnx_tests/test_lstm.py:149]: add api_2=api_2

File: tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py [hunk 5]
```diff
@@ -59,37 +62,50 @@ def create_tf_random_uniform_net(self, global_seed, op_seed, x_shape, min_val, m
...
         if ie_device == 'GPU':
             pytest.skip("RandomUniform is not supported on GPU")
-        self._test(*self.create_tf_random_uniform_net(**params, precision=precision, ir_version=ir_version,
-                                                      use_new_frontend=use_new_frontend), ie_device,
-                   precision, temp_dir=temp_dir, ir_version=ir_version, use_new_frontend=use_new_frontend, **params)
+        self._test(
+            *self.create_tf_random_uniform_net(**params, precision=precision, ir_version=ir_version,
+                                               use_new_frontend=use_new_frontend), ie_device,
+            precision, temp_dir=temp_dir, ir_version=ir_version, use_new_frontend=use_new_frontend,
+            **params)
```
Comment (dkrupnov) [inline tests/layer_tests/tensorflow_tests/test_tf_RandomUniform.py:111]: add api_2=api_2

**Review by dkrupnov — CHANGES_REQUESTED** [review]

**Review by dkrupnov — CHANGES_REQUESTED** [review]

**Review by dkrupnov — APPROVED** [review]

**Comment by rnugmanx** [comment]
Is there CI results?

**Comment by ishariko** [comment]
CI results: https://openvino-ci.intel.com/job/private-ci/job/customJobs/job/e2e-tests-linux/1633/
pre-commit: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/812

**Review by just-sparta — APPROVED** [review]

**Review by ValentinDymchishin — APPROVED** [review]

**Comment by ishariko** [comment]
UPD:
CI results: https://openvino-ci.intel.com/job/private-ci/job/developer-services/job/e2e/2099/
pre-commit: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/812

**Review by lidiaToropova — APPROVED** [review]

**Pull request merged**